### ISSUE TITLE
When a property is deprecated show the alternative

### ISSUE THREAD

**Issue by lordofthejars** [quarkusio/quarkus#33877]
### Description

I am using Quarkus Quartz, and I am getting the following warning: 

`2023-06-07 15:27:09,966 WARN  [io.qua.con.build] (main) The 'quarkus.quartz.start-mode' config property is deprecated and should not be used anymore`

It will also be nice to provide the new one, so I can change it.

PS: which is the new property?

**Comment by gsmet** [comment]
I think you need to use [`quarkus.scheduler.start-mode`](https://quarkus.io/guides/scheduler-reference#quarkus-scheduler_quarkus.scheduler.start-mode) now.

As for the warning, I agree, I'm just not sure it's easy to get this information but maybe we should add an annotation for it.

/cc @radcortez

**Comment by gsmet** [comment]
Yeah that's it, it's documented in the migration guide: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0#schedulerquartz .

**Comment by radcortez** [comment]
I think that a good place to keep this information is in javadoc `@deprecated` tag, which actually has the new property information already:

https://github.com/quarkusio/quarkus/blob/a00497a0ea1a2b2adba7b378fb121adf0e246450/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java#L71

With a new annotation just to state the new property, is more verbose and you need to be aware that the annotation exists, to add it and for it to work properly.

It seems to me that using the javadoc is the best option, but we do need a way to communicate it to the user. We could either:

- Record the `@deprecated` note and add it as part of the message
- Generate a link to the all configuration page, with the property name, add it to the message, which will open the javadoc information for that particular property.

Any thoughts?
